src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -215,21 +216,43 @@ public unsafe void Start()
                 _receivePause = new ManualResetEventSlim(true);
                 _reachability.StatusChanged += ReachabilityChanged;
                 _reachability.Start();
+
                 // STEP 1: Create the TcpClient, which is responsible for negotiating
                 // the socket connection between here and the server
-                try {
-                    // ReSharper disable once UseObjectOrCollectionInitializer
-                    _client = new TcpClient(AddressFamily.InterNetworkV6);
-                } catch (Exception e) {
-                    DidClose(e);
-                    return;
-                }
 
-                try {
-                    _client.Client.DualMode = true;
-                } catch (ArgumentException) {
-                    WriteLog.To.Sync.I(Tag, "IPv4/IPv6 dual mode not supported on this device, falling back to IPv4");
-                    _client = new TcpClient(AddressFamily.InterNetwork);
+                if (!String.IsNullOrEmpty(_options.NetworkInterface)) {
+                    try {
+                        var localAddress = GetLocalNIAddress(_options.NetworkInterface);
+                        if (localAddress == null) {
+                            WriteLog.To.Sync.I(Tag, $"Unknown Network Interface {_options.NetworkInterface}.");
+                            DidClose(new CouchbaseNetworkException(C4NetworkErrorCode.UnknownHost));
+                            return;
+                        }
+
+                        IPEndPoint localEndPoint = new IPEndPoint(localAddress, 0);
+                        var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+                        socket.Bind(localEndPoint);
+                        _client = new TcpClient() { Client = socket };
```

Comment:
TcpClient looks like it can take IPEndPoint inside the constructor, which I think would be more appropriate.

Reply:
I would guess that the answer is yes.  The latter might be doing some needless internal processing to find an appropriate address first in the parameterless constructor.